test/controllers/offer.controller.test.js

```diff
@@ -1444,297 +841,823 @@ describe('Offer Controller', () => {
+    });
+
+    itReturnsEmptyValuesWhenNoItemExistInDatabase({
+      endpoint,
+      method,
+      user: adminUser,
+      useExistingUser: true,
+    });
+
+    itReturnsEmptyValuesWhenNoItemExistInDatabase({
+      endpoint,
+      method,
+      user: vendorUser,
+      useExistingUser: true,
+    });
+
+    describe('when offers exist in db', () => {
+      beforeEach(async () => {
+        await Property.insertMany([...userProperties, ...user2Properties]);
+        await Enquiry.insertMany([...userEnquiries, ...user2Enquiries]);
+        await Offer.insertMany([...userOffers, ...user2Offers]);
+      });
+
+      itReturnsTheRightPaginationValue({
+        endpoint,
+        method,
+        user: regularUser,
+        useExistingUser: true,
+      });
+
+      itReturnsForbiddenForInvalidToken({ endpoint, method, user: editorUser });
+
+      itReturnsForbiddenForNoToken({ endpoint, method });
+
+      context('when request is sent by admin token', () => {
+        it('returns 26 offers', (done) => {
           request()
-            .put('/api/v1/offer/resolve-concern')
-            .set('authorization', userToken)
-            .send(concern)
+            [method](endpoint)
+            .set('authorization', adminToken)
             .end((err, res) => {
-              expect(res).to.have.status(403);
-              expect(res.body.success).to.be.eql(false);
-              expect(res.body.message).to.be.eql('You are not permitted to perform this action');
-              expect(sendMailStub.callCount).to.eq(0);
+              expect(res).to.have.status(200);
+              expect(res.body.success).to.be.eql(true);
+              expect(res.body.pagination.currentPage).to.be.eql(1);
+              expect(res.body.pagination.limit).to.be.eql(10);
+              expect(res.body.pagination.total).to.be.eql(26);
+              expect(res.body.pagination.offset).to.be.eql(0);
+              expect(res.body.result.length).to.be.eql(10);
               done();
             });
         });
       });
 
-      context('without token', () => {
-        it('returns error', (done) => {
+      context('when request is sent by vendor2 token', () => {
+        it('returns 8 offers', (done) => {
           request()
-            .put('/api/v1/offer/resolve-concern')
-            .send(concern)
+            [method](endpoint)
+            .set('authorization', vendor2Token)
             .end((err, res) => {
-              expect(res).to.have.status(403);
-              expect(res.body.success).to.be.eql(false);
-              expect(res.body.message).to.be.eql('Token needed to access resources');
-              expect(sendMailStub.callCount).to.eq(0);
+              expect(res).to.have.status(200);
+              expect(res.body.success).to.be.eql(true);
+              expect(res.body.pagination.currentPage).to.be.eql(1);
+              expect(res.body.pagination.limit).to.be.eql(10);
+              expect(res.body.pagination.total).to.be.eql(8);
```

Comment:
Lets try to use the length of the offers for most of the tests instead of writing the number. Lets fix this for related tests too.
```suggestion
              expect(res.body.pagination.total).to.be.eql(user2Offers.length);
```